core/src/main/java/hudson/Util.java

```diff
@@ -367,10 +363,8 @@ public static boolean isSymlink(@Nonnull File file) throws IOException {
     @SuppressWarnings("NP_BOOLEAN_RETURN_NULL")
     private static Boolean isSymlinkJava7(@Nonnull File file) throws IOException {
         try {
-            Object path = File.class.getMethod("toPath").invoke(file);
-            return (Boolean) Class.forName("java.nio.file.Files").getMethod("isSymbolicLink", Class.forName("java.nio.file.Path")).invoke(null, path);
-        } catch (NoSuchMethodException x) {
-            return null; // fine, Java 6
+            Path path = file.toPath();
+            return Files.isSymbolicLink(path);
         } catch (Exception x) {
```

Comment:
The code wouldn't be equivalent to what it was before, which was my first goal.

So, I say why not since that method is `private` so we know where and how it's used, but I'd prefer to do it in another PR in need be, so that if we need to bisect it later there's no functional change expected here in this commit.